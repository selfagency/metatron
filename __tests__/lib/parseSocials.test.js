const parseSocials = require('../../dist/lib/parseSocials.js').default

test('valid parseSocials', () => {
  const socials = [
    'https://facebook.com/profile',
    'https://twitter.com/profile',
    'https://linkedin.com/in/profile'
  ]

  expect(parseSocials(socials)).toMatchSnapshot()
})

test('invalid parseSocials', () => {
  const socials = 'socials'

  expect(() => parseSocials(socials)).toThrow()
})

test('invalid parseSocials', () => {
  const socials = [
    'facebook.com/profile',
    'twitter.com/profile',
    'linkedin.com/in/profile'
  ]

  expect(() => parseSocials(socials)).toThrow()
})
