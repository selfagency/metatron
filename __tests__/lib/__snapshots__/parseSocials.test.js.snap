// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid parseSocials 1`] = `
Array [
  Object {
    "service": "facebook",
    "url": "https://facebook.com/profile",
  },
  Object {
    "service": "twitter",
    "url": "https://twitter.com/profile",
  },
  Object {
    "service": "linkedin",
    "url": "https://linkedin.com/in/profile",
  },
]
`;
